test_linked_list.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+
+
+def test_node_class():
+    """Test Node."""
+    from linked_list import Node
+    new_node = Node('caterpillar')
+    assert new_node.val == 'caterpillar'
+
+
+def test_search():
+    """Search for node in list and return node."""
+    from linked_list import LinkedList
+    fake_list = LinkedList()
+    fake_list.insert('that')
+    assert fake_list.search('that') == 'that'
+
+
+def test_size():
+    """Test if size of list is returned."""
+    from linked_list import LinkedList
+    my_list = LinkedList()
+    my_list.insert('something')
+    my_list.insert('something else')
+    my_list.insert('Oh what the hell, one more thing.')
+    assert my_list.size() == 3
+
+
+def test_display():
+    """Print the list."""
+    from linked_list import LinkedList
+    my_list = LinkedList()
+    my_list.insert(1)
+    my_list.insert(2)
+    assert my_list.size() == len(my_list.display())
```

Comment:
Unit tests should avoid having dependencies. What if your .size() method changes at some point? Can you think of a better way to ensure your .display() method is working?